lib/adal.js

```diff
@@ -1153,16 +1153,31 @@ var AuthenticationContext = (function () {
      */
     AuthenticationContext.prototype._getResourceFromState = function (state) {
         if (state) {
-            var splitIndex = state.indexOf('|');
+            var splittedInfo = state.split('|');
 
-            if (splitIndex > -1 && splitIndex + 1 < state.length) {
-                return state.substring(splitIndex + 1);
+            if (splittedInfo.length > 1) {
+                return splittedInfo[1];
             }
         }
 
         return '';
     };
 
+    /**
+     * Extracts token expiration base time value from state.
+     * @ignore
+     */
+    AuthenticationContext.prototype._getExpireBaseTimeFromState = function (state) {
+        if (state) {
+            var splittedInfo = state.split('|');
+
+            if (splittedInfo.length > 2) {
+                return parseInt(splittedInfo[2], 10);
+            }
+        }
+        return 0;
```

Comment:
Could this instead default to the current time (which would simplify the downstream logic)?

Reply:
fixed as suggested